### PR TITLE
[bdd-engine] Support dynamic vars in lower camel and lower hyphen cases

### LINKED ISSUE #1347: Support dynamic vars in camel case and kebab case formats
Add support for camel case and kebab case dynamic variables, currently there are few camel case vars like responseCode and few kebab case like clipboard-text

### PR DESCRIPTION
**Pull request by uarlouski**
Closes #1347

### CODE REVIEW

File: vividus-tests/src/main/resources/story/integration/HTTP.story [hunk 1]
```diff
@@ -5,7 +5,7 @@ Meta:
 
 Scenario: Verify method DEBUG is supported; accidental space is trimmed in httpMethod enum
 When I issue a HTTP DEBUG  request for a resource with the URL 'http://example.org/'
-Then `${responseCode}` is equal to `405`
+Then `${response-code}` is equal to `405`
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/HTTP.story:8]: Add additional validation with a hyphen?

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java [hunk 2]
```diff
@@ -33,7 +36,16 @@
     public VariableResolver(IBddVariableContext bddVariableContext, Map<String, DynamicVariable> dynamicVariables)
     {
         this.bddVariableContext = bddVariableContext;
-        this.dynamicVariables = dynamicVariables;
+        this.dynamicVariables = new HashMap<>(dynamicVariables.size() * 2);
+        dynamicVariables.forEach((key, variable) ->
+        {
+            this.dynamicVariables.put(key, variable);
+            if (key.contains("-"))
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java:43]:
```java
if (key.indexOf('-') > -1)
```

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java [hunk 3]
```diff
@@ -33,7 +36,16 @@
     public VariableResolver(IBddVariableContext bddVariableContext, Map<String, DynamicVariable> dynamicVariables)
     {
         this.bddVariableContext = bddVariableContext;
-        this.dynamicVariables = dynamicVariables;
+        this.dynamicVariables = new HashMap<>(dynamicVariables.size() * 2);
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java:39]: let java do its work: if you have a single-word variable, then the size won't be x2

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java (hunk 3 above)
Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java:39]: but at the same time it won't be 16 buckets for 1 item

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java (hunk 2 above)
Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java:43]: benchmarks please :D

**Review by valfirst — APPROVED** [review]

**Pull request merged**